### PR TITLE
Introduce path mapping classes

### PR DESCRIPTION
**Pull request by rwols**
Path maps can be defined for each server configuration.
There is still the issue of actually applying the path maps everywhere. I will
defer this to another PR because it's huge.

I will also add the settings in sublime-package.json once everything is
actually working.

Related (but doesn't resolve): #535

### CODE REVIEW

**Comment by rwols** [comment]
Note that this could also have been a workaround for the broken vue language server by translating `C:` to `c:` :)

File: plugin/core/types.py [hunk 1]
```diff
@@ -621,6 +678,27 @@ def match_view(self, view: sublime.View) -> bool:
...
+                if mapped:
+                    break
+        return urllib.parse.urljoin('file:', urllib.request.pathname2url(path))
+
+    def map_server_uri_to_client_path(self, uri: str) -> str:
+        if os.name == 'nt':
+            # url2pathname does not understand %3A (VS Code's encoding forced on all servers :/)
+            path = urllib.request.url2pathname(urllib.parse.urlparse(uri).path).strip('\\')
+        else:
+            path = urllib.request.url2pathname(urllib.parse.urlparse(uri).path)
```
Comment (rchl) [inline plugin/core/types.py:694]: I wonder what cases it's trying to handle. Some tests would be nice to have to understand it better. But I guess Windows tests would have to run on Windows?

Comment (rwols) [reply plugin/core/types.py:694]: I took this directly from our good ol' `uri_to_filename` function in plugin/core/url.py

Comment (rchl) [reply plugin/core/types.py:694]: Why are we not using it directly then? :)

Comment (rwols) [reply plugin/core/types.py:694]: This is meant to replace `uri_to_filename` and `filename_to_uri` eventually. But for the time being I'll reuse the functions.

**Review by rchl — APPROVED** [review]

**Pull request merged**